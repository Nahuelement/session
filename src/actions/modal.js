import { types } from "../types/types"




export const openModal =() =>({
    type : types.activeModal,
    
})
export const endModal =() =>({
    type : types.endModal,
    
})