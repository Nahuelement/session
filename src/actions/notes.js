import { db } from "../firebase/fireConfig";
import { types } from "../types/types";
import { collection ,addDoc, doc, setDoc, deleteDoc} from "firebase/firestore";
import { loadNotes } from "../helpers/loadNotes";
import Swal from "sweetalert2";
import { fileUpload } from "../helpers/fileUpload";

// import cloudinary from 'cloudinary'

// // react-journal

//  cloudinary.config({ 
//     cloud_name: 'nahuelement', 
//     api_key: '723161997253356', 
//     api_secret: 'c2WpIaroeOWZNhdCdN8S0CQp-WE',
//     secure: true
//  });

export const startNewNotes = () =>{

    return async(dispatch, getState) =>{ // getState para obtener el estado

        const {uid} = getState().auth;
        

        const newNote = {
            title: '', 
            body: '',
            date: new Date().getTime(),
            url: ''
        }
        try{
            const docRef = await addDoc(collection(db,`${uid}/journal/notes`),newNote)
            dispatch(activeNote(docRef.id,newNote))
            // dispatch(startLodingNotes(uid)) IDEA MIA 
            dispatch(addNewNote(docRef.id,newNote))

        }
        catch (err) {
            console.log(err)
        } 
    }
}

export const activeNote = (id, note) => ({

    type:types.notesActive,
    payload:{
        id,
        ...note
    }
})

export const addNewNote = (id,note) => ({

    type:types.notesAddNew,
    payload:{
        id,
        ...note
    }
})


export const startLodingNotes = (uid) => {

    return async(dispatch) =>{

        try{
            const notes = await loadNotes(uid)
            dispatch(setNotes(notes))
        }
        catch(err) {
            console.log(err)

        }        
    }
}



export const setNotes = (notes) => ({

    type:types.notesLoad,
    payload:notes
    
}) 

export const startSaveNote = (note) => { 

    return async (dispatch, getState) =>{

        const {uid} = getState().auth; 
        
         if(!note.url){
             delete note.url;
         } 

        const noteToFirestore = {...note};
        delete noteToFirestore.id
        // console.log(noteToFirestore)
        
        try{
            await setDoc(doc(db,`${uid}/journal/notes/${note.id}`),noteToFirestore)
            Swal.fire('Guardado!',note.title,'success')
            dispatch(refreshNote(note.id,{...note}))
        }
        catch(err) {
            console.log(err)
        }
        
        


        


       
            
        
            
       
       
    }   

}
export const refreshNote = (id,note) =>({
    type:types.notesUpdated,
    payload:{
        id,
        note
    }

})

export const startUpLoading = (file) =>{
    return async(dispatch, getState) =>{

        const {active: activeNote} = getState().notes;
        
        Swal.fire({ 
            title: 'Uploading...',
            text:' Plase wait...',
            allowOutsideClick:false,
            allowEnterKey: () =>{
                Swal.showLoading();
            }
        }) 

        let fileUrl  = await fileUpload(file)
        
        Swal.close()
         if(!fileUrl){
         fileUrl= 'www.koi.partedelaprueba.porproblemasconJest().com'
        }
        activeNote.url = fileUrl
        console.log(activeNote)
       

        dispatch(startSaveNote(activeNote))
        
    }
}

export const startDeleting = (id) => {
    return async(dispatch, getState) =>{

        const uid = getState().auth.uid

        try{
            await deleteDoc(doc(db,`${uid}/journal/notes/${id}`)) 
            dispatch(deleteNotes(id))
        }
        catch(err) {
            console.log(err)
            
        }
        

        
    }
}

export const deleteNotes = (id) => ({
    type:types.notesDelete,
    payload:id
})

export const noteLogout = () => ({
    type:types.notesLogoutCleaning,
})