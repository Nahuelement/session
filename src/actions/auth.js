import { googleAuthProvider,facebookAuthProvider} from "../firebase/fireConfig"
import { getAuth, signInWithPopup, GoogleAuthProvider ,signInWithEmailAndPassword,
         FacebookAuthProvider,createUserWithEmailAndPassword,updateProfile,
         signOut} from "firebase/auth";
import { types } from "../types/types"
import { uiFinishLoading, uiStartLoading } from "./ui";
import Swal from "sweetalert2";
import { noteLogout } from "./notes";




export const startLogin = (email, password) => {

    return(dispatch) =>{
        dispatch(uiStartLoading())


        signInWithEmailAndPassword(getAuth(), email, password)
        .then(({user}) =>{
            dispatch(login(user.uid,user.displayName));
            dispatch(uiFinishLoading())
       
        }).catch(e => {
            
            // const errorCode = e.code;
            const errorMessage = e.message;

            let message = errorMessage;

            if(String(errorMessage)==='Firebase: Error (auth/invalid-email).'){
                 message = 'ingrese email valido'

            }else if (String(errorMessage)==='Firebase: Error (auth/wrong-password).'){
                 message = 'Password incorrecto'
                 
            }else if (String(errorMessage)==='Firebase: Error (auth/internal-error).'){
                 message = 'Ingrese email y contraseÃ±a'

                 
            }else if (String(errorMessage)==='Firebase: Error (auth/user-not-found).'){
                message = 'Email no registrado'
                
           }
           
            dispatch(uiFinishLoading())

            Swal.fire('Error',message,'error')
            
      })

    
}
}

export const startRegisterWhitEmailPassword  = (email, password,name) => {

    return(dispatch) =>{

    createUserWithEmailAndPassword(getAuth(),email, password)
    .then(async({user}) =>{
        
        await updateProfile(getAuth().currentUser,{displayName:name})
        
        dispatch(login(user.uid,user.displayName))
       
        
    })
    .catch((error) => {
        // const errorCode = error.code;
        const errorMessage = error.message;

        let message;
        if(String(errorMessage)==='Firebase: Error (auth/email-already-in-use).'){
            message = 'Usuario ya registrado con este email'

        }
       
        Swal.fire('Error',message,'error')

    }
    )
}}

export const starGoogleLogin = () => {
    
    return(dispatch) => {
        
       
        signInWithPopup(getAuth(),googleAuthProvider)
        .then(({user}) =>{
            dispatch(login(user.uid,user.displayName))
            
        }).catch((e) =>{
            
            GoogleAuthProvider.credentialFromError(e);
            
        })
        
    
    }}

    export const starFacebookLogin = () => {
    
        return(dispatch) => {
           
            signInWithPopup(getAuth(),facebookAuthProvider)
            .then((result) =>{
                const user = result.user;

                console.log(user)
            }).catch(FacebookAuthProvider.credentialFromError)
        }}
            

  

export const login = (uid, displayName) => ({

   
        type: types.login,
        payload: {
            uid,
            displayName
        }
})
export const startLogout = () => {

   return async(dispatch)=> {

    await signOut(getAuth())
    dispatch(logout())
    dispatch(noteLogout())

   }
    
}

export const logout = () => ({

    type: types.logout,
})
