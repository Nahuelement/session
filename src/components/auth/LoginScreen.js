import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import {  starGoogleLogin, startLogin } from '../../actions/auth'
import { useForm } from '../../hooks/useForm'



export const LoginScreen = () => {

    const dispatch = useDispatch()//PUEDE HACER DISPATCH DESDE CUALQUIER LUGAR DE LA APLICACION

    const {loading} = useSelector(state =>state.ui)
    const [formValues, HandleInputChange] = useForm({
        email: '',
        password: ''
    });

    const {email , password} = formValues;

    const handleLogin = () => {
        
        
        dispatch(startLogin(email,password))
      

    }
    const handleLoginGoogle = (e) => {
        
        dispatch(starGoogleLogin())

    }
    



    return (
        <>
        
            <h3 className="auth__title ">Login Mock</h3>
            <form onSubmit={handleLogin}>

                <input 
                type="text"
                placeholder="mail"
                name="email"
                className="auth__input"
                autoComplete = 'off'
                value = {email}
                onChange={HandleInputChange}
                />
                <input 
                type="password"
                placeholder="password"
                name="password"
                className="auth__input"
                autoComplete = 'off'
                value = {password}
                onChange={HandleInputChange}
                />


                <button
                type="submit"
                className="btn btn-primary btn-block"
                onClick={handleLogin}
                
                disabled = {loading}
                
                
                >Login
                
                </button>

      
                <div className="auth__social-networks">
                    <p>
                        Login con Google
                    </p>
                    <div 
                        className="google-btn"
                        onClick={handleLoginGoogle}
                        disabled = {loading}
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                    <br/>
                    
                </div>

                {/* <Link 
                to="/auth/register"
                className="link"
                >
                    Crear nueva cuenta
                </Link> */}


            </form>
            
        </>
    )
}
