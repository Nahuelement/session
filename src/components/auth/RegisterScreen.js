import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'
import validator from 'validator'

import { useDispatch, useSelector } from 'react-redux'
import { removeError, setError } from '../../actions/ui'
import { startRegisterWhitEmailPassword } from '../../actions/auth'
import Swal from "sweetalert2";



export const RegisterScreen = () => {



    const dispatch = useDispatch()
    const {msgError} = useSelector(state =>state.ui)

    
    // useEffect(() => {

    //     if (msgError){
    //     Swal.fire('Error',String(msgError),'error')}
        
        
    // }, [dispatch])


    const [ formvalues, handleInputChange ] = useForm({
        name:'esto es ',
        email:'un@ejemplo',
        password:'de ',
        confirm:'de un register'
    })
    const {name, email, password,confirm} = formvalues



    const handlenRegister = (e) => {
        e.preventDefault()
        

        if(isFormValid()){
            dispatch(startRegisterWhitEmailPassword(email, password,name))
            
        }
            
        
        

        
        
    }

    const isFormValid = () => {
        if(name.trim().length ===0){
            dispatch(setError('Nombre de usuario requerido'))
            
            Swal.fire('Error','Nombre de usuario requerido','error')
            
            return false
        }else if (!validator.isEmail(email)){
            dispatch(setError('Email valido requerido'))
            Swal.fire('Error','Email valido requerido','error')
            
            return false

        }else if (password!==confirm){
            dispatch(setError('password y confirm no son identicos'))
            Swal.fire('Error','password y confirm no son identicos','error')
            
            return false
        }
        else if (password.length <= 5){
            dispatch(setError('Password minimo 6 caracter '))
            Swal.fire('Error','Password minimo 6 caracter','error')
            
            return false}
        else{
            dispatch(removeError())
        }
        

        return true

    }

    return (
        <>

            <h3 className="auth__title">Register mock</h3>
            <form onSubmit={handlenRegister}>
                
                
                
                {msgError &&
                 (<div className="auth__alert-error">
                    {msgError}
                </div>)
                } 
              

                <input 
                type="text"
                placeholder='name'
                name="name"
                className="auth__input"
                autoComplete = 'off'
                value = {name}
                onChange={handleInputChange}
                />
                <input 
                type="text"
                placeholder='mail'
                name="email"
                className="auth__input"
                autoComplete = 'off'
                value = {email}
                onChange={handleInputChange}
                />
                <input 
                type="password"
                placeholder='password'
                name="password"
                className="auth__input"
                autoComplete = 'off'
                value = {password}
                onChange={handleInputChange}
                />
                <input 
                type="password"
                placeholder='coconfirm password'
                name="confirm"
                className="auth__input"
                autoComplete = 'off'
                value = {confirm}
                onChange={handleInputChange}
                />


                <button
                type="submit"
                onClick= {handlenRegister}
                className="btn btn-primary btn-block mb-5"
                
                >Register
                
                </button>

    

                <Link 
                to="/auth/login"
                className="link "
                >
                    Â¿Ya estas registrado?
                </Link>


            </form>
            
        </>
    )
}
