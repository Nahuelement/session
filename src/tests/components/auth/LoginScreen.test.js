

import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store' 
import thunk from "redux-thunk"
import { LoginScreen } from "../../../components/auth/LoginScreen";
import {mount} from 'enzyme'
import { MemoryRouter } from 'react-router-dom';
import { starGoogleLogin, startLogin } from '../../../actions/auth';



jest.mock('../../../actions/auth',() => ({ //mock devuelve un objeto ()
    starGoogleLogin:jest.fn(),
    startLogin: jest.fn()
}))



const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = { // inicial estado de mock que remplaza al store de redux
    auth:{},
    notes:{
        active:{
            id:'ImTMfs5frDggDoYFXhnm',
            title:'hola',
            body:'body',
            date: new Date().getTime() // para ver el cambio en la base de datos 
        }
    },
    ui:{
        loading:false,
        msgError:null 
    }
}
let store =mockStore(initialState)//STORE NO UTILIZA LOS REDUCER
store.dispatch = jest.fn()

describe('Pruebas en el <LoginScreen />', () => {

    beforeEach(() => {
        jest.clearAllMocks();
        store =mockStore(initialState)
        store.dispatch = jest.fn()

    });

    const wrapper = mount(
    
        <Provider store = {store}>
            <MemoryRouter>
                <LoginScreen />
            </MemoryRouter>
        </Provider>
    )

    test('el wrapper debe hacer match con el snapshot ', () => {

        expect(wrapper).toMatchSnapshot()

        
    })
    test('debe disparar la accion  starGoogleLogin ', () => {

        wrapper.find('.google-btn').prop('onClick')();
        expect(starGoogleLogin).toHaveBeenCalled()


        
    })
    test('debe disparar la accion startLogin ', () => {
        wrapper.find('.btn-block').prop('onClick')();
        expect(startLogin).toHaveBeenLastCalledWith('','') 
        
    })
    
    
    
    


    
    
})

