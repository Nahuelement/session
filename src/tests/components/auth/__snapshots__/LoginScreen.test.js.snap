// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <LoginScreen /> el wrapper debe hacer match con el snapshot  1`] = `
Array [
  <h3
    className="auth__title "
  >
    Login
  </h3>,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
    <div
      className="auth__social-networks"
    >
      <p>
        Ingresa con redes socieales
      </p>
      <div
        className="google-btn"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            alt="google button"
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </div>
        <p
          className="btn-text"
        >
          <b>
            Sign in with google
          </b>
        </p>
      </div>
      <br />
    </div>
    <a
      className="link"
      href="/auth/register"
      onClick={[Function]}
    >
      Crear nueva cuenta
    </a>
  </form>,
]
`;
